htmresearch/frameworks/nlp/model_factory.py

```diff
@@ -35,6 +41,34 @@
 """
 
 
+
+class ClassificationModelTypes(object):
+  """ Enumeration of supported classification model types, mapping userland
+  identifier to constructor.  See createModel() for actual factory method
+  implementation.
+  """
+  CioWordFingerprint = ClassificationModelFingerprint
+  CioDocumentFingerprint = ClassificationModelFingerprint
+  Keywords = ClassificationModelKeywords
+  HTMNetwork = ClassificationModelHTM
+  DocumentFingerPrint = ClassificationModelDocumentFingerprint
+  cioword = ClassificationModelFingerprint
+  docfp = ClassificationModelDocumentFingerprint
+  htm = ClassificationModelHTM
+  keywords = ClassificationModelKeywords
+
+
+  @classmethod
+  def getTypes(cls):
+    types = {attr
```

Comment:
I think this is a clever use of Python introspection, but I agree with @BoltzmannBrain that it's a bit of a head scratcher at first. I suggest a few clear comments to explain the implementation (perhaps in getTypes). This way it can be easier to maintain down the road.